quiz-microsevice/data-access/repositories/client-repository.js

```diff
@@ -12,8 +14,34 @@ const createOne = async (name, clientId, clientSecretHash) => {
 
 const retrieveOneByClientId = async (clientId) => {
   const client = await Client.findOne({ clientId });
-
+  
   return client ? toDTO(client) : null;
 };
 
-module.exports = { createOne, retrieveOneByClientId };
+const retrieveAllClients = async () => {
+  
+  const clients = await Client.find().select('-clientSecretHash');
+  
+  return clients.length > 0 ? clients : [];
+  
+  
+};
+
+/* 
+Here I make GET request to get ObjectID, not clientId.
+So I'm using (findById) method, not findOne.
+*/
+const regenerateClientCredentials = async(id, clientId, clientSecretHash) => {
+  const client = await Client.findById(id);
+  if(!client){
+    // console.log('hello')
+    throw new NotExistError("There is no client with this ObjectID.");
+  };
+  const regenerateClient = await Client.updateOne( { _id: id }, { clientId, clientSecretHash } );
```

Comment:
Use

const client = await Client.findOneAndUpdate(
  { _id: id }, 
  { clientId, clientSecretHash },
  { new: true } // This option returns the updated document
);

So you don't need to do unnecessary query